File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.js
/**
 * @param {number[]} arr
 * @param {number} k
 * @return {number}
 */
var findKthPositive = function(arr, k) {
    let map = {}
    for (let i = 0; i < arr.length; i++){
        map[arr[i]] = i;
    }
    let count = 1;
    let result = 0;
    while (count <= arr.length + k) {
        if (count in map === false ) result++;
        if (result === k) return count;
        count++;
        
    }
};

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        map = {}
        for i in range(len(arr)):
            map[arr[i]] = i
        count = 1
        result = 0
        while count <= len(arr) + k:
            if not count in map:
                result += 1
            if result == k:
                return count
            count += 1